### ISSUE TITLE
Application deployment status

### ISSUE THREAD

**Issue by HobeXD** [wukong-m2m/wukong-darjeeling#169]
In current version of WuKong, there is no way to know if the application deployed by Master succeeds on device side. We need a mechanism between Master and WuDevice to check/veryfiy/acknowledge if an application is deployed or is running on the device.

**Comment by wycc** [comment]
The deploy window UI is broken. In theory, it should be able to tell us if
an application is deployed successfully.

The other issue is that we have no way to tell what is currently running in
the wukong. We should try to keep the last deployment record so that we can
show the mapping and deployment result in the UI in the future. For now,
the run time information is lost after we restart the Master or Gateway.
This is incorrect. In addition, the discovery result is not saved in the
database as well. Therefore, we need to do discovery every time we start a
new deployment. This may not be an big issue for a small network. For a
larget network, the discovery might be very expensive and even impossible.

Those are issues exists for a long time and should be fixed.

YC

Phone: 886-2-23053135
Mobile: 0919816155

2016-05-10 14:46 GMT+08:00 HobeXD notifications@github.com:

**Comment by chishengshih** [comment]
YC, 

How about adding a protocol between WuDevice and Master to allow WuDevie returning the status including running WuObject and link? Can this protocol resolve the issue?

Daniel
−−--------------------------------------------
Chi-Sheng (Daniel) Shih
NEWS Lab, National Taiwan University
E-Mail: cshih@csie.ntu.edu.tw

## Phone: +886-2-33664927 Fax: +886-2-33663778

**Comment by bltsai** [comment]
The device could dump the link table and component map back to the Master
via the protocol. Due to the incremental deployment, we could get back a
running FBP diagram that is not what master just deployed but somewhat a
merged graph. To know whether a graph with labeled vertices is a isomorphic
to a subgraph of retrieved graph, it may need some computation time if the
system already runs a complex FBP.
YC,

How about adding a protocol between WuDevice and Master to allow WuDevie
returning the status including running WuObject and link? Can this protocol
resolve the issue?

Daniel
−−--------------------------------------------
Chi-Sheng (Daniel) Shih
NEWS Lab, National Taiwan University
E-Mail: cshih@csie.ntu.edu.tw

## Phone: +886-2-33664927 Fax: +886-2-33663778

**Comment by wycc** [comment]
We can get the wuobject already. However, the link table is useless since
we can not construct the original FBP from it. It's better to remember the
FBP and trhe mapping result in the Master.

2016-05-10 17:20 GMT+08:00 Chi-Sheng (Daniel) Shih <notifications@github.com

**Comment by chishengshih** [comment]
BL,

Can you summarize the issues and present it on Friday? We need to assign students to fix the problem.  
- Knowing if an FBP is successfully deployed.
- Knowing the link table on each device
- Knowing the status of wuobject on each device 

Daniel

**Comment by wycc** [comment]
Additional items.
- Keep the discovery result in the database.
- Query the device when it is added in the UI so that we do not need to
  discovery the whole network when we add new device.
- Keep the recent running FBP so that we can mark it in the application
  list.

2016-05-11 8:32 GMT+08:00 Chi-Sheng (Daniel) Shih notifications@github.com
:
